display/text_display.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package display
+
+import (
+	"errors"
+)
+
+type CursorDirection int
+
+const (
+	// Constants for moving the cursor relative to it's current position.
+	//
+	// Move the cursor one unit back.
+	Backward CursorDirection = iota
+	// Move the cursor one unit forward.
+	Forward
+	Up
+	Down
+)
+
+type CursorMode int
+
+const (
+	// Turn the cursor Off
+	CursorOff CursorMode = iota
+	// Enable Underline Cursor
+	CursorUnderline
+	// Enable Block Cursor
+	CursorBlock
+	// Blinking
+	CursorBlink
+)
+
+// TextDisplay represents an interface to a basic character device. It provides
+// standard methods implemented by a majority of character LCD devices. Pixel
+// type displays can implement this interface if desired.
+type TextDisplay interface {
+	// Enable/Disable auto scroll
+	AutoScroll(enabled bool) (err error)
+	// Return the number of columns the display supports
+	Cols() int
+	// Clear the display and move the cursor home.
+	Clear() (err error)
+	// Set the cursor mode. You can pass multiple arguments.
+	// Cursor(CursorOff, CursorUnderline)
+	//
+	// Implementations should return an error if the value of mode is not
+	// mode>= CursorOff && mode <= CursorBlink
+	Cursor(mode ...CursorMode) (err error)
+	// Move the cursor home (MinRow(),MinCol())
+	Home() (err error)
+	// Return the min column position.
+	MinCol() int
+	// Return the min row position.
+	MinRow() int
+	// Move the cursor forward or backward.
+	Move(dir CursorDirection) (err error)
+	// Move the cursor to arbitrary position. Implementations should return an
+	// error if row < MinRow() || row > Cols(), or col < MinCol() || col > Cols()
+	MoveTo(row, col int) (err error)
+	// Return the number of rows the display supports.
+	Rows() int
+	// Turn the display on / off
+	Display(on bool) (err error)
+	// return info about the display.
+	String() string
+	// Write a set of bytes to the display.
+	Write(p []byte) (n int, err error)
+	// Write a string output to the display.
+	WriteString(text string) (n int, err error)
+}
+
+type Intensity int
```

Comment:
What's the expected valid range for the value? Is it worth standardizing or is it better to leave the range to be device-specific? I don't have an opinion.

Reply:
I think most devices have a range of 0-255, but the increment can vary.

Reply:
I think it would be good to comment this explicitly. This can be done in a follow up.